tests/post/steps/test_authentication.py

```diff
@@ -0,0 +1,232 @@
+import json
+import re
+
+import requests
+import requests.exceptions
+
+import pytest
+from pytest_bdd import scenario, given, then, when, parsers
+
+import kubernetes.client
+from kubernetes.client.rest import ApiException
+
+from tests import kube_utils
+from tests import utils
+
+
+# Constants {{{
+
+INGRESS_PORT = 8443
+
+# }}}
+
+
+# Scenarios {{{
+
+@scenario('../features/authentication.feature', 'List Pods')
+def test_list_pods(host):
+    pass
+
+
+@scenario('../features/authentication.feature', 'Expected Pods')
+def test_expected_pods(host):
+    pass
+
+
+@scenario('../features/authentication.feature', 'Reach the OpenID Config')
+def test_reach_openid_config(host):
+    pass
+
+
+@scenario('../features/authentication.feature', 'Access HTTPS service')
+def test_access_https_service(host):
+    pass
+
+
+@scenario('../features/authentication.feature',
+          'Login to Dex using incorrect email')
+def test_failed_login(host):
+    pass
+
+
+@scenario('../features/authentication.feature',
+          'Login to Dex using correct email and password')
+def test_login(host):
+    pass
+
+
+# }}}
+
+
+# Fixtures {{{
+
+@pytest.fixture(scope='function')
+def context():
+    return {}
+
+
+@pytest.fixture(scope='function')
+def control_plane_ip(host):
+    with host.sudo():
+        output = host.check_output(' '.join([
+            'salt-call', '--local', '--out=json',
+            'grains.get', 'metalk8s:control_plane_ip',
+        ]))
+        ip = json.loads(output)['local']
+    return ip
+
+
+# }}}
+
+
+# When {{{
+
+
+@when(parsers.parse(
+    "we perform a request on '{path}' with port '{port}' on control-plane IP"))
+def perform_request(host, context, control_plane_ip, path, port):
+    def _get_urlpath_resource():
+        try:
+            context['response'] = requests.get(
+                'https://{ip}:{port}{path}'.format(
+                    ip=control_plane_ip, port=port, path=path
+                ),
+                verify=False,
+            )
+        except requests.exceptions.ConnectionError as exc:
+            pytest.fail(
+                "Failed to access oidc url path with error: {}".format(exc)
+            )
+    utils.retry(_get_urlpath_resource, times=10, wait=3)
```

Comment:
I think there shouldn't be a `ConnectionError` here, can't we wait for `ingress` to be ready?